### PR TITLE
improve $item handling in display arrays

### PR DESCRIPTION
**Pull request by qbalukom**
Fix externalLink as we have a talk on the daily.
Changes proposal:
- remove `textFormula`,
- change `linkFormula` to `link`,
- add icon to `externalLinks`,
- align docs.

### CODE REVIEW

File: core-ui/src/components/Extensibility/components/ExternalLink.js [hunk 1]
```diff
@@ -38,7 +43,13 @@ export const ExternalLink = ({ value, schema, structure, ...props }) => {
         target="_blank"
         rel="noopener noreferrer"
       >
-        {textFormula ? jsonata(textFormula) : value}
+        {value}
+        <Icon
+          glyph="inspect"
+          size="s"
+          className="fd-margin-begin--tiny"
+          ariaLabel="This link will be opened in a new tab"
```
Comment (wchrapka) [inline core-ui/src/components/Extensibility/components/ExternalLink.js:51]: shouldn't it be in the translation file?

File: docs/extensibility/display-section.md [hunk 2]
```diff
@@ -270,19 +270,17 @@ ExternalLink widgets render the link to an external page.
 
 #### Widget-specific parameters
 
-- **linkFormula** - an optional JSONata function to generate a custom link.
-- **textFormula** - an optional string or item which is displayed as a link.
+- **link** - an optional JSONata function to generate a custom link.
```
Comment (wchrapka) [inline docs/extensibility/display-section.md:273]: if this is optional, can we describe the default behavior?

**Review by wchrapka — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**